src/abstract_tree/mod.rs

```diff
@@ -0,0 +1,246 @@
+use egui::Rect;
+
+use crate::{
+    window::WindowState, Node, NodeIndex, NodeTree, Split, SurfaceIndex, TabDestination, TabIndex,
+};
+
+/// The basis for egui_dock
+/// This tree starts with a collection of surfaces, that then breaks down into nodes, and then into tabs.
+/// Indexing it will yield the NodeTree inside the surface which was indexed (panicing if it doesn't exist)
+pub struct Tree<Tab> {
```

Comment:
I don't think it makes a lot of sense to call it `Tree` anymore. It's a collection of trees with some additional state.

How about we call this `DockState` and rename `NodeTree` to just `Tree`?

Reply:
fixed in https://github.com/Adanos020/egui_dock/pull/149/commits/59213c2104013dbf956078c04138508915e0ece3